protected-api-access-protected-api/README.md

```diff
@@ -0,0 +1,142 @@
+| **Name**                          | `Node Web API`                                                               | Suggested value for this sample. <br/> You can change the app name at any time.                    |
+| **Supported account types**       | **Accounts in this organizational directory only (Single tenant)**           | Suggested value for this sample.                                                                   |
+| **Platform type**                 | _None_                                                                      | No redirect URI required; don't select a platform.                                                                    |
+| **Client secret**                 | _**Value** of the client secret (not its ID)_                                | :warning: Record this value immediately! <br/> It's shown only _once_ (when you create it).        |
+| **API Permissions**               | `user_impersonation`                                                         | Create a new permission called user_impersonation.  Required value for this sample.      |
+
+
+> :information_source: **Bold text** in the tables above matches (or is similar to) a UI element in the Azure portal, while `code formatting` indicates a value you enter into a text box in the Azure portal.
+
+
+
+### 2. Update code sample with app registration values
+
+```javascript
+  auth: {
+    // 'Application (client) ID' of app registration in Azure portal - this value is a GUID
+    clientId: '',
+    
+    // Client secret 'Value' (not the ID) from 'Client secrets' in app registration in Azure portal
+    clientSecret: '',
+    
+    // Full directory URL, in the form of https://login.microsoftonline.com/<tenant>
+    authority: ''
+  }
+```
+
+```javascript
+  secret: jwks.expressJwtSecret({
+    // Full URL, in the form of: https://login.microsoftonline.com/<tenant>/discovery/v2.0/keys
+    jwksUri: ''
+  }),
+```
+
+
+### 3. Install package(s)
+
+To install Node.Js and MSAL libraries:
+
+```bash
+npm install
+```
+
+## Run the application
+
+```bash
+node app.js
+```
+
+## Browse to the application
+
+Using postman or a similar app, navigate to **http://localhost:8080/me**
+You will need to provide an HTTP GET request such as:
+
+Authorization: Bearer {token}
+
+
+If everything worked, the sample app should produce the JSON-formatted output of a graph /me request for your user.
+
+```console
+{
+  "@odata.context":"https://graph.microsoft.com/v1.0/$metadata#users/$entity",
+  "businessPhones":[],
+  "displayName":"Firstname Lastname",
+  "givenName":"",
+  "jobTitle":null,
+  "mail":null,
+  "mobilePhone":null,
+  "officeLocation":null,
+  "preferredLanguage":null,
+  "surname":"Lastname",
+  "userPrincipalName":"",
+  "id":"00000000-0000-0000-0000-000000000000"
+  }
+```
+
+
+
+## About the code
+
+This demo shows how a confidential client application in Node.Js can call a protected web API on-behalf-of the initiating user.
+
```

Comment:
```suggestion
This Node.js application uses the Express web framework. The app has a single route that requires an access token. The access token will be automatically validated by MSAL:

- A missing or invalid (expired, wrong audience, etc) token will result in a `401` response.
- An otherwise valid token without the proper scope will result in a `403` response.
- A valid token with the proper scope of `user_impersonation` will be accepted, and the token will then be exchanged in an on-behalf-of OAuth flow to call to Microsoft Graph. The results of the Microsoft Graph call are then returned as the results of the API call.
```

Reply:
this one should be resolved